File: erp_bot/src/oip/modules/conversation/application/hygiene_gate_policy.py
"""PR-B6 / ADR_0089 — hygiene gate (NEXT-H1/H2 subset)."""

from __future__ import annotations

import json
from functools import lru_cache
from pathlib import Path
from typing import Any, Mapping

AUTHORITY = "ADR_0089"
STEP = "PR-B6"
DECISION = "PROD_READY_HYGIENE_GATE"
GAP_P1_005_STATUS = "REDUCED"
GAP_P2_004_STATUS = "REDUCED"


def _repo_root() -> Path:
    return Path(__file__).resolve().parents[6]


def registry_path() -> Path:
    return _repo_root() / "docs" / "mokxya-ai" / "MAI_HYGIENE_GATE_REGISTRY.json"


def run_status_path() -> Path:
    return _repo_root() / "artifacts" / "prod-ready-pr-b6" / "RUN_STATUS.json"


@lru_cache(maxsize=1)
def load_hygiene_gate_registry() -> dict[str, Any]:
    data = json.loads(registry_path().read_text(encoding="utf-8"))
    if data.get("authority") != AUTHORITY:
        raise RuntimeError("HYGIENE_REGISTRY_AUTHORITY_MISMATCH")
    if data.get("decision") != DECISION:
        raise RuntimeError("HYGIENE_DECISION_MISMATCH")
    return data


def load_run_status() -> dict[str, Any]:
    return json.loads(run_status_path().read_text(encoding="utf-8"))


def hygiene_gate_observability() -> dict[str, Any]:
    reg = load_hygiene_gate_registry()
    run = load_run_status()
    return {
        "hygiene_gate_step": STEP,
        "hygiene_gate_adr": AUTHORITY,
        "hygiene_gate_decision": reg["decision"],
        "gap_p1_005_register_status": GAP_P1_005_STATUS,
        "gap_p2_004_register_status": GAP_P2_004_STATUS,
        "invoice_print_syntax_fixed": True,
        "full_tsc_green_claimed": False,
        "playwright_required_green": False,
        "vacuous_greens_allowed": False,
        "engineering_pack_ready": bool(run.get("engineering_pack_ready")),
        "production_approved": False,
        "is_execution_authority": False,
    }


def assert_hygiene_gate_honesty(claim: Mapping[str, Any] | None = None) -> None:
    reg = load_hygiene_gate_registry()
    run = load_run_status()
    honesty = reg.get("honesty") or {}
    if honesty.get("production_approved") is True or run.get("production_approved") is True:
        raise RuntimeError("HYGIENE_PRODUCTION_APPROVED")
    if honesty.get("full_tsc_green_claimed") is True or run.get("full_tsc_green_claimed") is True:
        raise RuntimeError("FULL_TSC_GREEN_FALSE_CLAIM")
    if honesty.get("vacuous_greens_allowed") is True:
        raise RuntimeError("VACUOUS_GREENS_FORBIDDEN")
    if reg["gap_p2_004"].get("closed") is True and not run.get("full_tsc_green_claimed"):
        # CLOSED only if inventory cleared or signed green — not this ship
        raise RuntimeError("GAP_P2_004_FALSE_CLOSED")
    if not claim:
        return
    if claim.get("production_approved") is True:
        raise RuntimeError("HYGIENE_PRODUCTION_APPROVED")
    if claim.get("full_tsc_green_claimed") is True:
        raise RuntimeError("FULL_TSC_GREEN_FALSE_CLAIM")
    if claim.get("vacuous_greens_allowed") is True:
        raise RuntimeError("VACUOUS_GREENS_FORBIDDEN")
    if claim.get("gap_p2_004_closed") is True:
        raise RuntimeError("GAP_P2_004_FALSE_CLOSED")

File: src/platform/hygiene/hygieneGatePolicy.ts
/** PR-B6 / ADR_0089 — hygiene gate constants. */

export const HYGIENE_GATE_ADR = "ADR_0089" as const;
export const HYGIENE_GATE_STEP = "PR-B6" as const;
export const HYGIENE_GATE_DECISION = "PROD_READY_HYGIENE_GATE" as const;
export const INVOICE_PRINT_SYNTAX_FIXED = true;
export const FULL_TSC_GREEN_CLAIMED = false;
export const PLAYWRIGHT_REQUIRED_GREEN = false;
export const VACUOUS_GREENS_ALLOWED = false;
export const GAP_P1_005_REGISTER_STATUS = "REDUCED" as const;
export const GAP_P2_004_REGISTER_STATUS = "REDUCED" as const;
export const PRODUCTION_APPROVED = false;

export function hygieneGateSnapshot() {
  return {
    authority: HYGIENE_GATE_ADR,
    step: HYGIENE_GATE_STEP,
    decision: HYGIENE_GATE_DECISION,
    invoicePrintSyntaxFixed: INVOICE_PRINT_SYNTAX_FIXED,
    fullTscGreenClaimed: FULL_TSC_GREEN_CLAIMED,
    playwrightRequiredGreen: PLAYWRIGHT_REQUIRED_GREEN,
    vacuousGreensAllowed: VACUOUS_GREENS_ALLOWED,
    gapP1005RegisterStatus: GAP_P1_005_REGISTER_STATUS,
    gapP2004RegisterStatus: GAP_P2_004_REGISTER_STATUS,
    productionApproved: PRODUCTION_APPROVED,
  };
}
